Update babel to v7

macklinu opened the pull request:
This PR updates from babel v6 to v7 in the `server/` project.

I automated most of the upgrade using [babel-upgrade](https://github.com/babel/babel-upgrade) by running `npx babel-upgrade --write` in the `server/` directory. I will leave inline comments for anything that wasn't automated by that tool. 👍 

Some screenshots of running `npm run start-dev` to show the web client is still working on a basic level.

![image](https://user-images.githubusercontent.com/2344137/45522377-8418a900-b777-11e8-9adc-159f646c5fb8.png)

![image](https://user-images.githubusercontent.com/2344137/45522395-9e528700-b777-11e8-837a-167ad9430961.png)

Let me know if I can perform any other manual tests or if you have other questions!

TODO:

- [x] Fix test runner in `client/` directory -- d380cd1

macklinu commented on server/package.json:
```diff
@@ -14,14 +14,15 @@
-    "babel-plugin-transform-object-rest-spread": "^6.26.0",
-    "babel-preset-env": "^1.6.1",
-    "babel-preset-react": "^6.24.1",
+    "babel-eslint": "^9.0.0",
```
I manually updated to babel-eslint v9, as that uses babel v7 under the hood (check out the [release notes](https://github.com/babel/babel-eslint/releases/tag/v9.0.0)).

macklinu commented on server/.babelrc:
```diff
@@ -1,27 +1,32 @@
+        },
+        "debug": false,
+        "loose": true,
+        "useBuiltIns": "entry"
```
This used to be `"useBuiltIns": true`, but `true` is no longer a valid option for `@babel/preset-env` (see the [docs](https://babeljs.io/docs/en/babel-preset-env#usebuiltins)).

I chose `"entry"` here, which looks to inline parts of the `@babel/polyfill` that are required by the "browsers" option above (if I understand correctly).

Let me know if you have a preferred option here that isn't `"entry"` (again, the [docs](https://babeljs.io/docs/en/babel-preset-env#usebuiltins) for reference).

macklinu commented on server/package.json:
```diff
@@ -14,14 +14,15 @@
+    "@babel/core": "^7.0.0",
+    "@babel/plugin-proposal-class-properties": "^7.0.0",
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
+    "@babel/polyfill": "^7.0.0",
```
See the above comment about `"useBuiltIns"`. I think if we use it, we need to include the `@babel/polyfill` dependency too? I wasn't 100% sure about this one, so let me know if you have input on this.

macklinu commented:
Alright, the build is passing now - ready for revivew. 👓

tcbyrd commented:
@macklinu Thanks again for helping us improve Smee!

> Let me know if I can perform any other manual tests

A basic test would be sending Smee a webhook payload. We run this test on the server, but if you want to see it working in the browser you can send [this example payload](https://github.com/probot/smee/blob/master/server/tests/fixtures/issues.opened.json) to the channel (e.g. to a POST to `http://localhost:3000/:channel-name`)

I'll pull this down and review it as well. I'm hacking on Smee already, so shouldn't take very long.

macklinu commented:
still relevant - needs a final review

gr2m commented:
Pinning the issue as many folks are out due to holidays already, so it might take a while before someone can review

JasonEtco commented:
I'm actually around quite a bit more during the holiday season, as I can finally give my non-work projects some attention. @macklinu - its been a while  since these commits were first made, are there _even newer_ versions of these dependencies? If so, would you mind updating them one more time (and this time I'll review in a more timely manner)?

macklinu commented:
@jasonetco for sure! let me rebase this against master, ensure everything is up to date, and i’ll ping you when that’s done ✅

macklinu commented:
@JasonEtco this is ready for another review when you're free. Rebased against master, updated all Babel dependencies, reverted some `.babelrc` formatting, and force-pushed up. Some screenshots to confirm I can still `npm run start-dev`:

![image](https://user-images.githubusercontent.com/2344137/50059538-f74a0680-013c-11e9-960f-8beef8dea877.png)

![image](https://user-images.githubusercontent.com/2344137/50059540-fca75100-013c-11e9-8a6b-12ad72b0bd5f.png)
